packages/compose/src/higher-order/pure/test/index.js

```diff
@@ -24,6 +31,7 @@ describe( 'pure', () => {
 
 		// Updating with same props doesn't rerender.
 		rerender( <MyComp /> );
+		expect( console ).toHaveWarned();
```

Comment:
I always prefer `toHaveWarnedWith()` and a specific message, otherwise, we could easily have a bug that logs another warning and have the test fail to catch it.